docs: Update roadmap with v1.0 requirements (#356)
WorldHealthOrganization/app: docs/ROADMAP.md

  * Currently investigating options to integrate, including COVID Check
  * Ideally, this will be a fast follow-on
* Translation into wide variety of languages
  * Although we will shoot for as many of WHO's 6 official languages as possible, the requirement is only for English + one additional language
  * We will want to thoroughly vet any translations that we include in a public release
* Localization of content based on country
  * We may address this technically anyway based on our solution to content localization, but it is not a requirement for v1.0

#### Non-Technical Requirements

* Content spec finalized
* Design spec finalized
* Legal deliverables
  * Addendum to privacy policy for analytics
  * Legal review, including ToS/EULA if required
* Updated visual and text assets for app store listings

### vNext backlogs

This section is a major WORK IN PROGRESS and is not fully complete. Listing out very initial thoughts for now.

**GitHub Milestone:** https://github.com/WorldHealthOrganization/app/milestone/3

#### Fast Follow-Ons

Features that we will want in upcoming milestones quickly after v1.0 release:

* Triage survey integration

> ## Release Notes

Reviewer: I don't think Release Notes should be in a roadmap.  They should be in CHANGELOG.md.

Author: copy that, thought it'd be easier for this, but I'll pull into a separate file

Version numbering terminology:

* **v0.1**: refers to the initial buildout that will be submitted to the app stores but not intended for public release, aiming for submission on Thursday, March 26, 2020 ([v0.1 milestone](https://github.com/WorldHealthOrganization/app/milestone/2))
* **v1.0**: public release on the app stores, aiming for submission to app stores on Monday, March 30, 2020 ([v1.0 milestone](https://github.com/WorldHealthOrganization/app/milestone/4))
* **vNext**: describes features considered for future releases, but not yet prioritized into a specific milestone ([vNext milestone](https://github.com/WorldHealthOrganization/app/milestone/3))

> ### v0.1

Reviewer 2: v0.1 should be moved to the Release Notes and marked "not publicly released"

You can also use GitHub's release process with notes, for example: https://github.com/google/j2objc/releases - but that's better suited to a library that's pulled in as a dependency. For this, I think a single file with release notes seems more appropriate. Not sure where the file should be and how it should be named. Maybe just move it to the bottom of the ROADMAP?

Author: I've put this at the bottom of ROADMAP for now, we can move it out to a separate Release Notes file in the future or use the tags page

> **GitHub Milestone:** https://github.com/WorldHealthOrganization/app/milestone/2

Reviewer 2: A few lines earlier, add **Release Date:** Thursday, March 27th.... so we met our target!

  * License Agreement
* Account information/access from WHO to submit app on behalf of their accounts (also needed for code signing) ([#20](https://github.com/WorldHealthOrganization/app/issues/20) :warning:)

#### Other Notes

* We switched to the Flutter client framework for 0.1, and going forward.

### v1.0

**Goal:** Match or exceed the existing functionality in the WHO WhatsApp Bot presented idiomatically with native look-and-feel, with dynamically-updatable content localized to at least one other language besides English and ability to deliver push notifications to users.

> **Target Date:** Monday, March 30, 2020 - app store submission by EOD PDT

Reviewer: ```suggestion
**Target Date:** Monday, March 30, 2020 - transmission by team to WHO for app store approval by EOD PDT
```

**GitHub Milestone:** https://github.com/WorldHealthOrganization/app/milestone/4

#### Technical Features / Requirements

* Content parity with the [WHO WhatsApp Bot](https://api.whatsapp.com/send?phone=41225017615&text=hi&source=&data=)
* Native look-and-feel, designed to present content on mobile screens
* Localized content translated to at least one other language besides English
* Offline functionality with default content baked into the app bundle

> * Dynamically updated content served from the cloud so users don't need to update the app to receive content updates

Reviewer: ```suggestion
* Dynamically updated content served from the cloud so users don't need to update the app to receive content updates
* Instrumentation to understand how users are using the app
```

Author: is this not captured in the Analytics bullet? happy to use "instrumentation" terminology if it's preferred, I just wanted to be very specific with what we're capturing

Reviewer: No you're right.   I mixed two concepts.  The Q is whether it is a requirement to have *other than* user-behavior analytics (crash reports for example).  I'll remove this comment.

* Localized push notification capabilities
  * _(Needs Discussion)_ What are the localization requirements for v1.0? What level of granularity will we need in region (e.g. language, continent, country, city, etc)? How will the user choose their location(s) for push delivery?

> * Analytics capabilities with the ability to collect aggregated data for product improvement complying with all applicable laws & regulations and app policies

Reviewer: ```suggestion
* Analytics capabilities with the ability to understand aggregated data about user behavior for product improvement complying with all applicable laws & regulations and app policies
```

The analytics system would still be *collecting* information from each user, but we are simply *using* it in aggregated form.

Author: @advayDev1 I updated this line to take into account both of your points - hopefully this covers crash reporting too

> * _(Needs Discussion)_ Distribution pipeline using [fastlane](https://fastlane.tools/) for integration with WHO's existing app distribution process

Reviewer: ```suggestion
* _(Needs Discussion)_ Distribution pipeline using [fastlane](https://fastlane.tools/) for integration with WHO's existing app distribution process
* Full security review
```

>   * License Agreement

Reviewer: ```suggestion
```

> * Account information/access from WHO to submit app on behalf of their accounts (also needed for code signing) ([#20](https://github.com/WorldHealthOrganization/app/issues/20) :warning:)

Reviewer: Remove as this isn't what happened.
```suggestion
```

Reviewer 2: Future plan is to setup Fastlane and have that handle app store submission

#### Other Notes

* We switched to the Flutter client framework for 0.1, and going forward.

### v1.0

**Goal:** Match or exceed the existing content in the WHO WhatsApp Bot presented idiomatically with native look-and-feel, with dynamically-updatable content localized to at least one other language besides English and ability to deliver push notifications to users.

**Target Date:** Monday, March 30, 2020 - transmission of app to WHO team for app store submission by EOD PDT

**GitHub Milestone:** https://github.com/WorldHealthOrganization/app/milestone/4

#### Technical Features / Requirements

* Content parity with the [WHO WhatsApp Bot](https://api.whatsapp.com/send?phone=41225017615&text=hi&source=&data=)
* Native look-and-feel, designed to present content on mobile screens
* Localized content translated to at least one other language besides English
* Offline functionality with default content baked into the app bundle

> * Dynamically updated content served from the cloud so users don't need to update the app to receive content updates

Reviewer 3: Is this in v1.0 or vNext? The relevant issue (https://github.com/WorldHealthOrganization/app/issues/24) is currently part of the vNext milestone.

Author: This is v1.0 - good catch, I've updated the ticket! I think I need to do some general ticket cleanup this evening

* Localized push notification capabilities
  * _(Needs Discussion)_ What are the localization requirements for v1.0? What level of granularity will we need in region (e.g. language, continent, country, city, etc)? How will the user choose their location(s) for push delivery?

> * Instrumentation capabilities for understanding aggregated useage information to be used for product improvement, complying with all applicable laws & regulations and app policies

Reviewer 3: ```suggestion
* Instrumentation capabilities for understanding aggregated usage information to be used for product improvement, complying with all applicable laws & regulations and app policies
```

> * _(Needs Discussion)_ Distribution pipeline using [fastlane](https://fastlane.tools/) for integration with WHO's existing app distribution process

Reviewer 2: Analytics should be included in there as a feature, very difficult to know what we're doing without that

Author: I'd used the "Instrumentation" terminology to include analytics, but I added analytics to that bullet point for readability 👍

* Full security review

#### Features NOT Included

* Triage survey advising users on actions to take based on their symptoms
  * Currently investigating options to integrate, including COVID Check
  * Ideally, this will be a fast follow-on
* Translation into wide variety of languages
  * Although we will shoot for as many of WHO's 6 official languages as possible, the requirement is only for English + one additional language
  * We will want to thoroughly vet any translations that we include in a public release
* Localization of content based on country
  * We may address this technically anyway based on our solution to content localization, but it is not a requirement for v1.0

#### Non-Technical Requirements

* Content spec finalized
* Design spec finalized
* Legal deliverables
  * Addendum to privacy policy for analytics

>   * Legal review

Reviewer: ```suggestion
  * Legal review, including ToS/EULA if required
```
